content/blog/2018-08-14-where-to-bird.md

```diff
@@ -0,0 +1,365 @@
+
+``` r
+indx_W <- which (dat$osm_polygons$natural %in% c ("water", "wetland"))
+indx_N <- which (!dat$osm_polygons$natural %in% c ("water", "wetland"))
+
+xy_W <- sf::st_coordinates (dat$osm_polygons [indx_W, ])
+xy_N <- sf::st_coordinates (dat$osm_polygons [indx_N, ])
+```
+
+Then we use Mark’s [`geodist`
+package](https://github.com/hypertidy/geodist) to get pairwise distances
+between all points, find minima, and make a data.frame to submit to
+`add_osm_surface()`. We have 7068 watery points and 10065 non-watery
+points so the speed of `geodist` is crucial here!
+
+``` r
+t1 <- Sys.time()
+d <- geodist::geodist (xy_W, xy_N)
+# so fast!!!
+Sys.time() - t1
+```
+
+    ## Time difference of 13.57983 secs
+
+``` r
+d1 <- apply (d, 1, min)
+d2 <- apply (d, 2, min)
+xy <- cbind (rbind (xy_W, xy_N), c (d1, d2))
+xy <- xy [, c (1, 2, 5)]
+colnames (xy) <- c ("x", "y", "z")
+xy <- xy [!duplicated (xy), ]
+```
+
+Finally we plot the results on the roads we had gotten earlier, although
+we do not recommend staying on the middle of a road to bird! We re-add
+the points corresponding to the MPI and bird hide after the surface
+coloring. With `osmplotr`, order matters because layers are added on top
+of each other. Note that plotting the distances takes a while.
+
+``` r
+# colorblind-friendly palette!
+cols <- viridis::viridis_pal (direction = -1) (30)
+
+add_osm_surface (map0, dat_H,
+                 dat = xy, col = cols) %>%
+    add_axes()  %>%
+  add_colourbar(cols = cols,
+                zlim = range(xy[,"z"])) %>%
+  add_osm_objects(mpi_sf, 
+                  col = 'white', size = 5) %>%
+  add_osm_objects(results$osm_points, 
+                  col = 'white', size = 5)%>%
+  print_osm_map (filename = 'map_a3.png', width = 600,
+                 units = 'px', dpi = 72)
+
+magick::image_read('map_a3.png') %>%
+  magick::image_annotate("Map data © OpenStreetMap contributors",
+                         color = "white",
+                         boxcolor =  "black",
+                         size = 15,
+                         gravity = "south")
+```
+
+<img src="/img/blog-images/2018-08-14-where-to-bird/unnamed-chunk-9-1.png" width="600" />
+
+On the map, the yellower/lighter a road is, the better it is to observe
+birds according to Mark’s assumption that birds are most likely to be
+found where water lies close to natural areas. With this metric, the MPI
+itself is not too badly located, which after all is not too surprising
+for an MPI of *ornithology*. Maybe one should just walk to the one of
+the nearest lakes to meet some birds.
+
+### Conclusion
+
+#### A note about open geographical data
+
+In this post we used `opencage`, `osmdata` and `osmplotr` therefore
+mosly making use of the awesome OpenStreetMap data (The OpenCage
+Geocoder uses a bit more, but it only warrants [attributing
+OpenStreetMap](https://opencagedata.com/faq#legal)). We therefore added
+the annotation “Map data © OpenStreetMap contributors” using rOpenSci’s
```

Comment:
say what you added the annotation to (i.e. the last map)

Reply:
ok